geminidr/core/primitives_preprocess.py

```diff
@@ -533,6 +501,147 @@ def dilateObjectMask(self, adinputs=None, suffix=None, dilation=1, repeat=False)
+            Use a local median filter for single pixels?
+        debug : bool
+            Display regions?
+
+        """
+        log = self.log
+        log.debug(gt.log_message("primitive", self.myself(), "starting"))
+        timestamp_key = self.timestamp_keys[self.myself()]
+
+        if regions is not None:
+            regions = regions.split(';')
+        elif regions_file is not None:
+            with open(regions_file) as f:
+                regions = f.read().strip().splitlines()
+        else:
+            raise ValueError('regions must be specified either as a string '
+                             '(regions) or with a file (regions_file)')
+
+        for ad in adinputs:
+            for ext in ad:
+                for region in regions:
+                    try:
+                        slices = cartesian_regions_to_slices(region.strip())
+                    except ValueError:
+                        log.warning(f'Failed to parse region: {region}')
+                        continue
+
+                    if len(slices) != ext.data.ndim:
+                        raise ValueError(f'region {region} does not match '
+                                         'array dimension')
+
+                    region_shape = np.array([s.stop - s.start for s in slices])
+
+                    # Find the axis that will be used for the interpolation
+                    if axis is None:
+                        # If we have two axis with the same size, we should
+                        # use the deeper one. E.g. for images with a square
+                        # region, interpolation is done on lines.
+                        use_axis = np.where(
+                            region_shape == region_shape.min())[0][-1]
+                    else:
+                        if axis not in range(ext.data.ndim):
+                            raise ValueError('axis should specify a dimension '
+                                             f'between 0 and {ext.data.ndim}')
+                        use_axis = axis
+
+                    if debug:
+                        log.debug(f'Replacing pixel {region} with a '
+                                  'local median ')
+                        plot_slices = [slice(sl.start - 10, sl.stop + 10)
+                                       for sl in slices]
+                        if len(plot_slices) > 2:
+                            plot_slices = [Ellipsis] + plot_slices[-2:]
+                        origdata = ext.data[plot_slices].copy()
+
+                    if use_local_median and np.prod(region_shape) == 1:
+                        nd = NDAstroData(data=ext.data,
+                                         mask=np.zeros(ext.shape, dtype=bool))
+                        nd.mask[slices] = 1
+                        ring_median_filter(nd, 3, 5, inplace=True,
+                                           replace_flags=1,
+                                           replace_func='median')
```

Comment:
Instead of creating a new (possibly large) `mask` array just to flag one pixel, would it be better to set a high bit of the "bad" pixel and pass that to the function?
```
ext.mask[slices] |= 32768
ring_median_filter(..., replace_flags=32768, ...)
ext.mask[slices] ^= 32768
```

Reply:
Yes, seems a good idea!